### ISSUE TITLE
Word2Vec wordsNearest issue

### ISSUE THREAD

**Issue by raver119** [deeplearning4j/deeplearning4j#968]
Word2Vec wordsNearest uses plain search against all vocab for similar words. And that's definitely non-always-desired implementation. It's ok if you search for similar words only once per execution, but for multiple executions, other solution would be nice to have.

TODO: VPTree-based implementation for wordsNearest. And it would be nice, if it won't eat whole system memory.

**Comment by raver119** [comment]
done, TreeModelUtils
